### PR TITLE
In 179 music update

### PR DESCRIPTION
**Pull request by rox51124**

### CODE REVIEW

File: src/AudioDevice.cpp [hunk 1]
```diff
@@ -22,6 +23,11 @@ indie::AudioDevice::~AudioDevice()
     CloseAudioDevice();
 }
 
+void indie::AudioDevice::initAudio()
+{
+    InitAudioDevice();
```
Comment (iDommel) [inline src/AudioDevice.cpp:28]: It would be better if there was a verification to check if audiodevice wasn't already initialized.

**Review by iDommel — CHANGES_REQUESTED** [review]

**Review by Varriss — DISMISSED** [review]

**Review by iDommel — APPROVED** [review]

**Review by iDommel — DISMISSED** [review]

File: src/systems/GameSystem.cpp [hunk 2]
```diff
@@ -37,7 +43,11 @@ namespace indie
 
         sceneManager.addScene(createScene(), SceneManager::SceneType::GAME);
         sceneManager.setCurrentScene(SceneManager::SceneType::GAME);
+<<<<<<< HEAD
```
Comment (Varriss) [inline src/systems/GameSystem.cpp:46]: Remove shit pls

File: src/systems/GameSystem.cpp [hunk 3]
```diff
@@ -67,9 +77,76 @@ namespace indie
...
+        std::shared_ptr<Component> component3 = std::make_shared<Component>();
+        std::shared_ptr<Entity> e = std::make_shared<Entity>();
+        std::shared_ptr<Rect> rect  = std::make_shared<Rect>(0, 0, 0, 0);
+        std::shared_ptr<Position> pos = std::make_shared<Position>(500, 500);
+        std::shared_ptr<Sprite> sprite = std::make_shared<Sprite>("test_pictures/scarfy.png", 6);
+        std::shared_ptr<Entity> e2 = std::make_shared<Entity>();
+        std::shared_ptr<Position> pos2 = std::make_shared<Position>(0, 0, 0);
+        std::shared_ptr<Model3D> model = std::make_shared<Model3D>("test_models/turret.obj", "test_models/turret_diffuse.png");
+
+=======
```
Comment (Varriss) [inline src/systems/GameSystem.cpp:98]: Shit

File: src/systems/GameSystem.cpp [hunk 4]
```diff
@@ -67,9 +77,76 @@ namespace indie
 
     std::shared_ptr<IEntity> GameSystem::createCamera(Vector3 camPos, Vector3 camTarget)
     {
+<<<<<<< HEAD
```
Comment (Varriss) [inline src/systems/GameSystem.cpp:80]: Shit

File: src/systems/GameSystem.cpp (hunk 3 above)
Comment (Varriss) [inline src/systems/GameSystem.cpp:98]: Shit

File: src/systems/GameSystem.cpp [hunk 5]
```diff
@@ -67,9 +77,76 @@ namespace indie
...
+        std::shared_ptr<Entity> e = std::make_shared<Entity>();
+        std::shared_ptr<Rect> rect  = std::make_shared<Rect>(0, 0, 0, 0);
+        std::shared_ptr<Position> pos = std::make_shared<Position>(500, 500);
+        std::shared_ptr<Sprite> sprite = std::make_shared<Sprite>("test_pictures/scarfy.png", 6);
+        std::shared_ptr<Entity> e2 = std::make_shared<Entity>();
+        std::shared_ptr<Position> pos2 = std::make_shared<Position>(0, 0, 0);
+        std::shared_ptr<Model3D> model = std::make_shared<Model3D>("test_models/turret.obj", "test_models/turret_diffuse.png");
+
+=======
+>>>>>>> origin/dev
```
Comment (Varriss) [inline src/systems/GameSystem.cpp:99]: Shit

File: src/systems/GameSystem.cpp [hunk 6]
```diff
@@ -67,9 +77,76 @@ namespace indie
...
         std::shared_ptr<Entity> cam = std::make_shared<Entity>();
         std::shared_ptr<CameraComponent> camera = std::make_shared<CameraComponent>(camTarget, camPos);
 
+<<<<<<< HEAD
```
Comment (Varriss) [inline src/systems/GameSystem.cpp:103]: Shit

File: src/systems/GameSystem.cpp [hunk 7]
```diff
@@ -67,9 +77,76 @@ namespace indie
...
+
+        e->addComponent(rect)
+            .addComponent(pos)
+            .addComponent(sprite);
+
+        e2->addComponent(pos2)
+            .addComponent(model);
+
+=======
+>>>>>>> origin/dev
```
Comment (Varriss) [inline src/systems/GameSystem.cpp:149]: Shit

File: src/systems/GameSystem.cpp [hunk 8]
```diff
@@ -93,6 +170,9 @@ namespace indie
         }
     }
 
+<<<<<<< HEAD
+        scene->addEntities({entity2, e, e2, cam, e3, e4, soundEntity});
+=======
```
Comment (Varriss) [inline src/systems/GameSystem.cpp:175]: Shit

File: src/systems/GameSystem.cpp [hunk 9]
```diff
@@ -93,6 +170,9 @@ namespace indie
         }
     }
 
+<<<<<<< HEAD
```
Comment (Varriss) [inline src/systems/GameSystem.cpp:173]: Shit

File: src/systems/GameSystem.cpp [hunk 10]
```diff
@@ -110,6 +190,7 @@ namespace indie
         createPlayer(*scene, KEY_RIGHT, KEY_LEFT, KEY_UP, KEY_DOWN, 1, {10, 0, 10});
         createPlayer(*scene, KEY_D, KEY_A, KEY_W, KEY_S, 2, {0, 0, 0});
         scene->addEntities({modelEntity, createCamera({50.0f, 50.0f, 50.0f}, {0.0f, 10.0f, 0.0f})});
+>>>>>>> origin/dev
```
Comment (Varriss) [inline src/systems/GameSystem.cpp:193]: Shit

File: src/components/SoundComponent.hpp [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    class SoundComponent : public Component
+    {
+    public:
+        SoundComponent(std::string str) : sound(str) { _type = IComponent::Type::SOUND; };
+
+        std::string getValue() const { return sound; };
+        void setSoundState(Sound::SoundState state) { _newState = state; };
+        Sound::SoundState getSoundState() { return _newState; };
+        Sound::SoundState _newState = Sound::SoundState::PLAYING;
+        std::string sound;
```
Comment (Varriss) [inline src/components/SoundComponent.hpp:26]: Transform to private

File: src/components/SoundComponent.hpp [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+namespace indie
+{
+    class SoundComponent : public Component
+    {
+    public:
+        SoundComponent(std::string str) : sound(str) { _type = IComponent::Type::SOUND; };
+
+        std::string getValue() const { return sound; };
+        void setSoundState(Sound::SoundState state) { _newState = state; };
+        Sound::SoundState getSoundState() { return _newState; };
```
Comment (Varriss) [inline src/components/SoundComponent.hpp:24]: Add const to getter methods

File: src/components/MusicComponent.hpp [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    class MusicComponent : public Component
+    {
+    public:
+        MusicComponent(std::string str) : music(str) { _type = IComponent::Type::MUSIC; };
+
+        std::string getValue() const { return music; };
+        void setMusicState(Music::MusicState state) { _newState = state; };
+        Music::MusicState getMusicState() { return _newState; };
+        Music::MusicState _newState = Music::MusicState::PLAY;
+        std::string music;
```
Comment (Varriss) [inline src/components/MusicComponent.hpp:27]: Transform to private

File: src/components/MusicComponent.hpp [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+{
+
+    class MusicComponent : public Component
+    {
+    public:
+        MusicComponent(std::string str) : music(str) { _type = IComponent::Type::MUSIC; };
+
+        std::string getValue() const { return music; };
+        void setMusicState(Music::MusicState state) { _newState = state; };
+        Music::MusicState getMusicState() { return _newState; };
```
Comment (Varriss) [inline src/components/MusicComponent.hpp:25]: Add const to getters methods

File: src/Sound.hpp [hunk 15]
```diff
@@ -51,10 +51,10 @@ namespace indie
          * @param volume Volume value
          */
         void setVolume(float volume);
+        SoundState _state = SoundState::PAUSED;
```
Comment (Varriss) [inline src/Sound.hpp:54]: Transform to Private, make a const getter

File: src/Music.hpp [hunk 16]
```diff
@@ -43,11 +43,13 @@ namespace indie {
             float getTimeLengthMusic();
             ///  @brief Get current music time played (in seconds)
             float getTimePlayedMusic();
+            /// @brief Check if music is playing
+            bool isMusicPlaying();
+            MusicState _state = MusicState::PAUSE;
```
Comment (Varriss) [inline src/Music.hpp:48]: Transform to private

**Review by Varriss — CHANGES_REQUESTED** [review]

**Review by iDommel — APPROVED** [review]

**Review by iDommel — APPROVED** [review]

**Pull request merged**